crates/wasi-common/src/wasi.rs

```diff
@@ -40,6 +40,7 @@ impl From<wiggle::GuestError> for Errno {
             InFunc { .. } => Self::Inval,
             InDataField { .. } => Self::Inval,
             SliceLengthsDiffer { .. } => Self::Fault,
+            BorrowCheckerOOM { .. } => Self::Fault,
```

Comment:
Is `BorrowCheckerOOM` meant to signify an out-of-memory error? If that's the case, I'd suggest we be more verbose and spell the full thing out here: `BorrowCheckerOutOfMemory`. It hopefully will be less confusing for the newcomers.

Reply:
`BorrowCheckerOutOfHandles` is the right way to spell it, thanks